presto-common/src/main/java/com/facebook/presto/common/block/BlockUtil.java

```diff
@@ -76,6 +77,13 @@ static void checkValidPosition(int position, int positionCount)
         }
     }
 
+    static void checkValidSliceRange(int sourceIndex, int length)
+    {
+        if (sourceIndex > MAX_ARRAY_SIZE - length) {
```

Comment:
nit. might be good to leave a comment as to why we are subtraction instead of addition. 
Something like
// souceIndex + length check can overflow . Hence we are subtracting length from MAX_ARRAY_SIZE .